### PR TITLE
sorting stream

### PR DESCRIPTION
**Pull request by nikob411**

### CODE REVIEW

File: exercises/sorting/sorted_stream.py [hunk 1]
```diff
@@ -10,4 +13,21 @@ def sorted_stream(*streams):
     :param streams: Sorted iterators
     :return: A generator which yields the sorted elements of all streams.
     """
-    pass
+#    for it in streams:
+#        for element in it:
+#             yield element
```
Comment (bharris47) [inline exercises/sorting/sorted_stream.py]: remove commented code

**Review by bharris47 — CHANGES_REQUESTED** [review]
Give me a few sentences of what a heap is and why it's appropriate for this problem.

File: exercises/sorting/sorted_stream.py (hunk 1 above)
Comment (nikob411) [reply exercises/sorting/sorted_stream.py]: ARRRRRITE

Comment (nikob411) [reply exercises/sorting/sorted_stream.py]: Heap data structure is mainly used to represent a priority queue. We use heapify to convert the iterable into a heap data structure. We then use heapreplace to have the first element of the data popped then then pushed. The value larger than the pushed value can be returned. Lastly, we use heappop to sort the elements and maintain the heap structure.

Comment (nikob411) [reply exercises/sorting/sorted_stream.py]: Would like to see your solution.

**Review by bharris47 — APPROVED** [review]

**Pull request merged**